### PR TITLE
feat(DLI): add a new resource to manage dli datasource connection

### LINKED ISSUE #1669: [DLI]Need a new resource to manage Datasource Connections
### Current Terraform Version

```
...
```

### Proposal

1. support manage  the enhanced datasource connection, the system will automatically create a VPC peering connection and required routes.
2. support manage  the basic datasource connection, the system will automatically create a VPC peering connection and required routes.

### References

https://support.huaweicloud.com/intl/en-us/usermanual-dli/dli_01_0410.html

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1669 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDatasourceConnection_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDatasourceConnection_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceConnection_basic
=== PAUSE TestAccDatasourceConnection_basic
=== CONT  TestAccDatasourceConnection_basic
--- PASS: TestAccDatasourceConnection_basic (306.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       306.826s
```

### CODE REVIEW

File: docs/resources/dli_datasource_connection.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+subcategory: "Data Lake Insight (DLI)"
+---
+
+# huaweicloud_dli_datasource_connection
+
+Manages a DLI datasource connection resource within HuaweiCloud.  
+
+## Example Usage
+
+```HCL
```
Comment (houpeng80) [inline docs/resources/dli_datasource_connection.md]: change to lowercase

File: docs/resources/dli_datasource_connection.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+subcategory: "Data Lake Insight (DLI)"
+---
+
+# huaweicloud_dli_datasource_connection
+
+Manages a DLI datasource connection resource within HuaweiCloud.
```
Comment (ShiChangkuo) [inline docs/resources/dli_datasource_connection.md]:
```suggestion
Manages a DLI datasource **enhanced** connection resource within HuaweiCloud.  
```

File: docs/resources/dli_datasource_connection.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+$ terraform import huaweicloud_dli_datasource_connection.test 0ce123456a00f2591fabc00385ff1234
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `tags`.
+It is generally recommended running `terraform plan` after importing a resource.
+You can then decide if changes should be applied to the resource, or the resource definition should be updated to
+align with the resource. Also you can ignore changes as below.
+
+```
```
Comment (ShiChangkuo) [inline docs/resources/dli_datasource_connection.md]:
```suggestion
```hcl
```

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+		return rst
+	}
+	return nil
+}
+
+func resourceDatasourceConnectionDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteDatasourceConnection: missing operation notes
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go]: this comment is hard to understand

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+	return resourceDatasourceConnectionRead(ctx, d, meta)
+}
+
+func addRoutes(cfg *config.Config, region string, id string, addRaws *schema.Set) error {
+	for _, params := range addRaws.List() {
+		var (
+			addConnectionRouteHttpUrl = "v2.0/{project_id}/datasource/enhanced-connections/{id}/routes"
+			addConnectionRouteProduct = "dli"
+		)
+		addConnectionRouteClient, err := cfg.NewServiceClient(addConnectionRouteProduct, region)
```
Comment (hmbig2) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go]: declare the client in the parameters.

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go [hunk 6]
```diff
@@ -0,0 +1,577 @@
...
+		err = addRoutes(cfg, region, d.Id(), v.(*schema.Set))
+		if err != nil {
+			return diag.Errorf("error adding routes to DatasourceConnection: %s", d.Id())
+		}
+	}
+
+	return resourceDatasourceConnectionRead(ctx, d, meta)
+}
+
+func buildCreateDatasourceConnectionBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go]: how about just remove config

File: docs/resources/dli_datasource_connection.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  Changing this parameter will create a new resource.
+
+* `queues` - (Optional, List) List of queue names that are available for datasource connections.
+
+* `hosts` - (Optional, List) The user-defined host information. A maximum of 20,000 records are supported.
+The [Host](#datasourceConnection_Host) structure is documented below.
+
+* `routes` - (Optional, List) List of routes.
+
+  Changing this parameter will create a new resource.
```
Comment (Jason-Zhang9309) [inline docs/resources/dli_datasource_connection.md]: remove this line

File: docs/resources/dli_datasource_connection.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dli_datasource_connection.md]: done

File: docs/resources/dli_datasource_connection.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/dli_datasource_connection.md]: done

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go]: removed

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection.go]: ok

File: docs/resources/dli_datasource_connection.md (hunk 7 above)
Comment (hmbig2) [reply docs/resources/dli_datasource_connection.md]: ok

**Pull request merged**